internal/cri/server/container_create.go

```diff
@@ -163,6 +163,11 @@ func (c *criService) CreateContainer(ctx context.Context, r *runtime.CreateConta
 
 	var volumeMounts []*runtime.Mount
 	if !c.config.IgnoreImageDefinedVolumes {
+		err = c.mutateMounts(ctx, config.GetMounts(), platform)
```

Comment:
I don't think the `mutateMounts` should be here, because it's to prepare OCI image volume instead of container image. It should apply for all the platforms before building container OCI spec.

REF: https://github.com/kubernetes/kubernetes/pull/125659

Reply:
summary..

Image(s) have been pulled with pod runtime handler annotation, but at that point we don't know if they are going to be used as a container or bind mounted image volumes at volume mount target paths inside the container rootfs. 

Then we get a container create that contains a set of 0-n "special" volume mounts that we need to "mutate" by 1) creating a bind mountable local volume using the pod/config selected runtime snapshotter... and lease scope that is cleaned up to allow the pod resources to clean up.. (at least until we come up with a way to define and collect at node level n pods..) 2) modify these special mounts to have the above created host path referring to the pod leased local volume path requested mounts for this container. 3) follow through allowing the mounts to be added to the container spec.  

Side note: there was some discussion about the new volume mount being read only and not executable?  have to check.

Reply:
Thanks for comment! Yes. For the alpha stage, we should limit the mount with readonly and noexec.